bundles/org.openhab.core.audio/src/main/java/org/openhab/core/audio/FileAudioStream.java

```diff
@@ -70,6 +75,36 @@ private static AudioFormat getAudioFormat(File file) throws AudioException {
         }
     }
 
+    private static AudioFormat parseWavFormat(File file) {
+        try (InputStream inputStream = new FileInputStream(file);
+                BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream);
+                AudioInputStream audioInputStream = AudioSystem.getAudioInputStream(bufferedInputStream);) {
+            javax.sound.sampled.AudioFormat format = audioInputStream.getFormat();
+            String javaSoundencoding = format.getEncoding().toString();
+            String codecPCMSignedOrUnsigned;
+            if (javaSoundencoding.equals(Encoding.PCM_SIGNED.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_SIGNED;
+            } else if (javaSoundencoding.equals(Encoding.PCM_UNSIGNED.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_UNSIGNED;
+            } else if (javaSoundencoding.equals(Encoding.ULAW.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_ULAW;
+            } else if (javaSoundencoding.equals(Encoding.ALAW.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_ALAW;
+            } else {
+                codecPCMSignedOrUnsigned = null;
+            }
```

Comment:
Can you use a `switch-case-statement` for this block?

Reply:
I don't think so ?
I already tried, but the value "javaSoundencoding.equals(Encoding.SOMETHING.toString()" are not constant and the compilation failed because you can only switch to constant value.
Unless I miss something.